### PR TITLE
Simplify directory structure

### LINKED ISSUE #874: Change jsonnet directory layout
**What is missing?**
Good user experience.

**Why do we need it?**
To improve user experience by categorizing various jsonnet snippets to create a clear picture.

**Environment**

* kube-prometheus version:

    post release-0.7

**Anything else we need to know?**:

This is building on top of https://github.com/prometheus-operator/kube-prometheus/pull/873

I would like to categorize files in `jsonnet/kube-prometheus` directory in a following pattern:
```
.
├── jsonnetfile.json
├── main.libsonnet # previously kube-prometheus.libsonnet
├── components/  # holding major components of the stack
│   ├── mixin/
│   │   ├── alerts
│   │   │   ├── alerts.libsonnet
│   │   │   ├── general.libsonnet
│   │   │   ├── node.libsonnet
│   │   │   └── tests.yaml
│   │   ├── custom.libsonnet
│   │   ├── kubernetes.libsonnet
│   │   └── rules
│   │       ├── general.libsonnet
│   │       ├── node-rules.libsonnet
│   │       └── rules.libsonnet
│   ├── alertmanager.libsonnet
│   ├── blackbox-exporter.libsonnet
│   ├── grafana.libsonnet
│   ├── ksm-autoscaler.libsonnet
│   ├── kube-state-metrics.libsonnet
│   ├── kube-rbac-proxy.libsonnet
│   ├── node-exporter.libsonnet
│   ├── prometheus.libsonnet
│   ├── prometheus-adapter.libsonnet
│   └── prometheus-operator.libsonnet
├── addons/  # various snippets that can be applied separatelly
│   ├── dropping-deprecated-metrics-relabelings.libsonnet
│   ├── grafana-weave-net-cluster.json
│   ├── grafana-weave-net.json
│   ├── kube-prometheus-all-namespaces.libsonnet
│   ├── kube-prometheus-anti-affinity.libsonnet
│   ├── kube-prometheus-bootkube.libsonnet
│   ├── kube-prometheus-config-mixins.libsonnet
│   ├── kube-prometheus-custom-metrics.libsonnet
│   ├── kube-prometheus-external-metrics.libsonnet
│   ├── kube-prometheus-insecure-kubelet.libsonnet
│   ├── kube-prometheus-ksonnet.libsonnet
│   ├── kube-prometheus-managed-cluster.libsonnet
│   ├── kube-prometheus-node-ports.libsonnet
│   ├── kube-prometheus-static-etcd.libsonnet
│   ├── kube-prometheus-strip-limits.libsonnet
│   ├── kube-prometheus-thanos-sidecar.libsonnet
│   └── kube-prometheus-weave-net.libsonnet
└── platforms/  # Ready to apply examples for each platform.
    ├── kube-prometheus-eks.libsonnet
    ├── kube-prometheus-gke.libsonnet
    ├── kube-prometheus-kops-coredns.libsonnet
    ├── kube-prometheus-kops.libsonnet
    ├── kube-prometheus-kubeadm.libsonnet
    ├── kube-prometheus-kube-aws.libsonnet
    └── kube-prometheus-kubespray.libsonnet
```

RFC @prometheus-operator/kube-prometheus-reviewers

### PR DESCRIPTION
**Pull request by paulfantom**
Closes #874

cc @prometheus-operator/kube-prometheus-reviewers

### CODE REVIEW

**Comment by paulfantom** [comment]
Files are moved as discussed in #874. I don't have that many environments to test all examples from `examples/` directory and I did what I could to make them functional however I expect there will be some bugs in there.

I am now focusing on writing some docs on how to transition to a new repo structure, but I will do this in a separate PR since this one is already starting to be quite large.

**Comment by paulfantom** [comment]
Ready for review. ping @prometheus-operator/kube-prometheus-reviewers

**Review by lilic — COMMENTED** [review]
the changes are very large to review, sadly right now do not have time for a focused review, so not blocking if someone gets a chance to actually have a look!

> I am now focusing on writing some docs on how to transition to a new repo structure, but I will do this in a separate PR since this one is already starting to be quite large.

I would personally prefer to wait for that docs PR, as it might help me understand this change. Do you think you can open that one as well? If there are too many upgrade problems or huge steps for our users to do ,we might want to hold off or change the direction of few things. 

Hope that makes sense?

Thanks!! 🎉

File: examples/additional-namespaces-servicemonitor.jsonnet [hunk 1]
```diff
@@ -1,11 +1,13 @@
...
+  values+:: {
+    common+: {
+      namespace: 'monitoring',
+    },
     prometheus+:: {
       namespaces+: ['my-namespace', 'my-second-namespace'],
     },
   },
-  prometheus+:: {
+  prometheus+: {
```
[the remark this answers is not in the document]
Comment (paulfantom) [reply examples/additional-namespaces-servicemonitor.jsonnet:10]: Yes. `prometheus` object is no longer hidden.

**Comment by paulfantom** [comment]
> I would personally prefer to wait for that docs PR, as it might help me understand this change.

Sure, I will open a PR with docs as soon as I finish writing it ;)

**Review by brancz — APPROVED** [review]
Since there are no changes in the output, lgtm.

I agree, UX can be discussed when we're done and documenting it.

**Pull request merged**